src/amo/components/AddonMoreInfo.js

```diff
@@ -0,0 +1,75 @@
+import React, { PropTypes } from 'react';
+import { compose } from 'redux';
+
+import Link from 'amo/components/Link';
+import translate from 'core/i18n/translate';
+
+import './AddonMoreInfo.scss';
+
+
+export class AddonMoreInfoBase extends React.Component {
+  static propTypes = {
+    addon: PropTypes.object.isRequired,
+    i18n: PropTypes.object.isRequired,
+  }
+
+  render() {
+    const { addon, i18n } = this.props;
+
+    if (!addon) {
+      return null;
```

Comment:
I'd like to see this check done in the parent instead:
````javascript
{addon ? <AddonMoreInfo addon={addon} /> : null}
````

Otherwise, what purpose would  `AddonMoreInfo` serve for `null` add-ons? Anyway, the `addon` property is declared to be required which I think is the right thing to do.